catroid/src/main/java/org/catrobat/catroid/ui/controller/RecentBrickListManager.java

```diff
@@ -0,0 +1,83 @@
+	/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.ui.controller;
+
+import org.catrobat.catroid.common.FlavoredConstants;
+import org.catrobat.catroid.common.RecentBricksHolder;
+import org.catrobat.catroid.content.bricks.Brick;
+import org.catrobat.catroid.io.RecentBrickListSerializer;
+
+import java.io.File;
+import java.util.List;
+
+import static org.catrobat.catroid.common.Constants.BACKPACK_DIRECTORY_NAME;
+import static org.catrobat.catroid.common.Constants.RECENT_BRICKS_FILE;
+import static org.catrobat.catroid.common.FlavoredConstants.DEFAULT_ROOT_DIRECTORY;
+
+public final class RecentBrickListManager {
+
+	private static final RecentBrickListManager INSTANCE = new RecentBrickListManager();
+
+	public final File recentBrickListDirectory =
+			new File(FlavoredConstants.DEFAULT_ROOT_DIRECTORY, BACKPACK_DIRECTORY_NAME);
+	public final File recentBricksFile = new File(recentBrickListDirectory, RECENT_BRICKS_FILE);
+
+	private RecentBricksHolder recentBricksHolder = new RecentBricksHolder();
+
+	private RecentBrickListSerializer recentBricksSerializer =
+			new RecentBrickListSerializer(recentBricksFile);
+
+	public static RecentBrickListManager getInstance() {
+		return INSTANCE;
+	}
+
+	private RecentBrickListManager() {
+		createBackpackDirectories();
+	}
+
+	private void createBackpackDirectories() {
+		DEFAULT_ROOT_DIRECTORY.mkdir();
+	}
+
+	public List<Brick> getRecentBricks() {
+		return recentBricksHolder.getRecentBricks();
+	}
+
+	public void addBrick(Brick brick) {
+		if (!recentBricksHolder.contains(brick)) {
```

Comment:
Please rethink the logic of this method. When a brick is already contained in the `recentBricksHolder` then remove it and add it again, such that it is displayed on top again - as the most recently used brick should always be on top.